core/src/types/operator/operator.rs

```diff
@@ -544,6 +544,59 @@ impl Operator {
         self.write_with(path, OpWrite::new(), bs).await
     }
 
+    /// Copy a file from `from` to `to`.
+    ///
+    /// # Notes
+    ///
+    /// - `from` and `to` must be a file.
+    /// - `to` will be overwritten if it exists.
+    /// - If `from` and `to` are the same, nothing will happen.
+    /// - `copy` is idempotent. For same `from` and `to` input, the result will be the same.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use std::io::Result;
+    /// # use opendal::Operator;
+    ///
+    /// # #[tokio::main]
+    /// # async fn test(op: Operator) -> Result<()> {
+    /// op.copy("path/to/file", "path/to/file2").await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn copy(&self, from: &str, to: &str) -> Result<()> {
+        if from == to {
+            return Ok(());
+        }
```

Comment:
Same above.

In a normal POSIX copy, if the from and to are the same, the contents of this file will simply be cleared. 
Do we want to keep this behavior?

Other backends don't necessarily behave this way. So I simply return directly.

Reply:
I prefer to return an `IsSameFile` error here.